### PR TITLE
fix: vtt validations

### PR DESCRIPTION
**Pull request by parodyBit**
- changed all TextFields to TextFormFields - reworked the validators to return String? instead of void. if it is valid it returns null, if it is not valid it returns an error message. - moved the state changes to the onChanged controllers in form
- added `WitValueFormatter` Max number of decimals is 9 Max whole-number is 10
- added `WitAddressFormatter` max length is 42

### CODE REVIEW

**Review by Tommytrg — COMMENTED** [review]
Some validations are directly related to Witnet's business logic. What do you think about moving that logic to a separate file so we can reuse it anywhere in the future? What I'd move to another file is:
- Witnet address validation
- wit value validation

File: lib/util/storage/database/balance_info.dart [hunk 1]
```diff
@@ -73,4 +73,26 @@ class BalanceInfo {
...
+    List<Utxo> selectedUtxos = utxoPool.cover(
+      amountNanoWit: value,
+      utxoStrategy: selectionStrategy,
+    );
+    if (selectedUtxos.isEmpty) return null;
+    int changeNanoWit;
+    int valuePaidNanoWit =
+        BalanceInfo.fromUtxoList(selectedUtxos).availableNanoWit;
+    changeNanoWit = (valuePaidNanoWit - value);
+    return changeNanoWit > 0
```
Comment (parodyBit) [inline lib/util/storage/database/balance_info.dart:94]: @aesedepece does this look correct?

File: lib/util/storage/database/balance_info.dart [hunk 2]
```diff
@@ -73,4 +73,26 @@ class BalanceInfo {
...
+    });
+    List<Utxo> selectedUtxos = utxoPool.cover(
+      amountNanoWit: value,
+      utxoStrategy: selectionStrategy,
+    );
+    if (selectedUtxos.isEmpty) return null;
+    int changeNanoWit;
+    int valuePaidNanoWit =
+        BalanceInfo.fromUtxoList(selectedUtxos).availableNanoWit;
+    changeNanoWit = (valuePaidNanoWit - value);
```
Comment (parodyBit) [inline lib/util/storage/database/balance_info.dart:93]: @aesedepece does this look correct to compute a weighted fee?

File: lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart [hunk 3]
```diff
@@ -43,27 +36,41 @@ class RecipientStepState extends State<RecipientStep>
...
+  FeeType _feeType = FeeType.Weighted;
   bool _showFeeInput = false;
+  final _addressController = TextEditingController();
+  final _addressFocusNode = FocusNode();
+  final _amountController = TextEditingController();
+  final _amountFocusNode = FocusNode();
+  final _minerFeeController = TextEditingController();
+  final _minerFeeFocusNode = FocusNode();
 
+  bool forceValidation = false;
```
Comment (gabaldon) [inline lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart:52]: I think this is not needed anymore

File: lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart [hunk 4]
```diff
@@ -72,140 +79,104 @@ class RecipientStepState extends State<RecipientStep>
...
+      _errorAddressText = _validateAddress(_address);
+      _errorAmountText = _validateAmount(_amount);
+      _hasAddressError = _errorAddressText != null;
+      _hasAmountError = _errorAmountText != null;
+      if (_feeType == FeeType.Absolute) {
+        _errorFeeText = _validateFee(_minerFee);
+        _hasFeeError = _errorFeeText != null;
+      }
+    });
+    if ((!_hasAddressError && !_hasAmountError && !_hasFeeError) &&
```
Comment (gabaldon) [inline lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart:171]: You can avoid using _hasAddressError, _hasAmountError and _hasFeeError 
```suggestion
    if ((_errorAddressText == null && _errorAmountText == null && _errorFeeText == null) &&
```

File: lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart (hunk 3 above)
Comment (parodyBit) [reply lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart]: removed!

File: lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart [hunk 5]
```diff
@@ -229,12 +193,9 @@ class RecipientStepState extends State<RecipientStep>
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
     final extendedTheme = theme.extension<ExtendedTheme>();
-    _addressFocusNode.addListener(() => _validateAddress());
-    _amountFocusNode.addListener(() => _validateAmount());
-    _minerFeeFocusNode.addListener(() => _validateFee());
     return Form(
       key: _formKey,
-      autovalidateMode: AutovalidateMode.onUserInteraction,
+      autovalidateMode: AutovalidateMode.disabled,
```
Comment (gabaldon) [inline lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart:198]: Realized this does not validate the form on clicking outside the input (on unfocus)

Comment (parodyBit) [reply lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart:198]: fixed!

**Review by gabaldon — APPROVED** [review]

**Pull request merged**